src/devices/cpu/cadr/cadr_dasm.h

```diff
@@ -0,0 +1,40 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+// ********************************************************************************
+// * MIT CADR processor microcode disassembler
+// ********************************************************************************
+
+#ifndef MAME_CPU_CADR_CADR_DASM_H
+#define MAME_CPU_CADR_CADR_DASM_H
+
+#pragma once
+
+class cadr_disassembler : public util::disasm_interface
+{
+public:
+	cadr_disassembler() = default;
+	virtual ~cadr_disassembler() = default;
+
+	virtual u32 opcode_alignment() const override;
+	virtual offs_t disassemble(std::ostream &stream, offs_t pc, const data_buffer &opcodes, const data_buffer &params) override;
+
+
+private:
+	static constexpr u64 NOP_MASK = u64(0x7fffffffeffff);
+
+	static const char *const bool_op[0x10];
+	static const char *const arith_op[0x10];
+	static const char *const arith_op_c[0x10];
+	static const char *const mult_div_op[0x20];
+	static const char *const output_bus_control[0x04];
+	static const char *const q_control[0x04];
+	static const char *const rp[0x04];
```

Comment:
Do these really need to be in the header?  It’s just exposed implementation details that slow down compiling downstream stuff and increase the chance of causing a recompile if the implementation changes.  Throw them in an anonymous namespace in the .cpp file.